### PR TITLE
Support Mocked Unique Jobs

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
@iwiznia 

Fixes: https://github.com/Expensify/Expensify/issues/68858

This adds support for unique jobs to mockRequests.

It does this by counting the state of `mockRequest` as part of what makes a job `unique`. Thus, if you run `CreateJob` twice with `unique` set, and the same name, but one has `mockRequest` and one doesn't, you get two jobs created, one with `mockRequest: true` in its data, and one without.

This means that in a 10x test scenario, where the first request is real and the remaining 9 are mocked and `unique` is set, the first request creates one (non-mocked) job. The second request creates a (mocked) job, and the remaining 8 requests return the second job (because they'd be duplicates of the existing unique job).

Tests:
Run `CreateJobTest::uniqueJob`  with 2x enabled, i.e.:
```
./test -only CreateJob -duplicateRequests 2
```

Watch the logs, and you . should see 6 `CreateJob` requests.

1. The first created a non-mocked job.
2. The second created a mocked job.
3. The third looks up and returns the ID of the first job.
4. The fourth looks up and returns the ID of the second job.
5. The fifth updates the first job and returns its ID.
6. The sixth updates the second job and returns its ID.

requests 1, 3, and 5 are not mocked. Requests 2, 4, and 6 are mocked.

### CODE REVIEW

File: plugins/Jobs.cpp [hunk 1]
```diff
@@ -253,18 +253,22 @@ bool BedrockPlugin_Jobs::peekCommand(SQLite& db, BedrockCommand& command) {
...
                 SQResult result;
-                SINFO("Unique flag was passed, checking existing job with name " << job["name"]);
+                SINFO("Unique flag was passed, checking existing job with name " << job["name"] << ", mocked? "
+                      << (command.request.isSet("mockRequest") ? "true" : "false"));
+                string operation = command.request.isSet("mockRequest") ? "IS NOT" : "IS";
                 if (!db.read("SELECT jobID, data "
                              "FROM jobs "
-                             "WHERE name=" + SQ(job["name"]) + ";",
+                             "WHERE name=" + SQ(job["name"]) +
+                             "  AND JSON_EXTRACT(data, '$.mockRequest') " + operation + " NULL;",
```
Comment (iwiznia) [inline plugins/Jobs.cpp]: I think we should time this to be sure we are not introducing a performance regression.

**Comment by tylerkaraszewski** [comment]
@iwiznia Timing info:

For the worst case scenario, the job we have the most instances of is `www-prod/UpdateDomainWorker`, with 746612 jobs queued (at the time of this message).

Running the following query:
```
SELECT jobID, data FROM jobs WHERE name='www-prod/UpdateDomainWorker';
```
Takes 2.5 seconds.

Updating it to check for whether it's mocked or not takes 27.8 seconds. This . sounds like a big regression, but this is not a unique job (or there wouldn't be 746,612) this is just a check if someone accidentally passed the `unique` flag along with the wrong job. This is slow, but I think acceptably slow for something that's not supposed to happen.
```
SELECT jobID, data FROM jobs WHERE name='www-prod/UpdateDomainWorker' AND JSON_EXTRACT(data, '$.mockRequest') IS NOT NULL;
```

When we run this against a job with only one instance (as should be the case for unique jobs), with this query:
```
SELECT jobID, data FROM jobs WHERE name='www-prod/ScrapeCards?accountID=5370327';
```
It returns instantly (0.003s).

And when we update it to include the check for mockRequest:
```
SELECT jobID, data FROM jobs WHERE name='www-prod/ScrapeCards?accountID=5370327' AND JSON_EXTRACT(data, '$.mockRequest') IS NOT NULL;
```
Timing is unaffected (still 0.003s).

**Comment by iwiznia** [comment]
Ummm, a bit confused, so what if we have thousands of jobs with different IDs and we check for one? So, let's say we have thousands of `www-prod/ScrapeCards?accountID=XXX` (with different accountIDs) and we try to create one more? In both cases in which it does and doesn't already exist.

**Comment by tylerkaraszewski** [comment]
Discussed in person. @iwiznia said he'd merge. :)

**Comment by iwiznia** [comment]
Totally forgot, sorry.

**Pull request merged**